preview-env/destroy/action.yml

```diff
@@ -0,0 +1,95 @@
+---
+name: Destroy preview environment
+
+description: Destroy preview environment.
+
+inputs:
+  app_name:
+    description: Argocd app name (for some projects the value is known ahead).
+    required: false
+    default: ""
+
+  deployment_ref:
+    description: >-
+      This is supposed to contain the branch name that will be used to lookup the actual Argo CD deployment name.
+      Typically it's populated using `github.event.ref`.
+    required: false
+    default: ${{ github.event.ref }}
+
+  revision:
+    description: Revision or branch of the Helm chart. Typically the branch to be deployed, or `master`.
+    required: true
+
+  argocd_server:
+    description: URL of the Argo CD instance to target.
+    required: false
+    default: argocd.int.camunda.com
+
+  argocd_token:
+    description: An Argo CD token with sufficient permissions to create Applications.
+    required: true
+
+  argocd_version:
+    default: v2.3.1
+    description: Version tag of Argo CD CLI tool
+    required: false
+runs:
+  using: composite
+  steps:
+  - name: Restore cache
+    uses: actions/cache@v3
+    id: tool-cache-argocd
+    with:
+      path: /usr/local/bin/argocd
+      key: ${{ runner.os }}-tool-cache-argo-${{ inputs.argocd_version }}
+  - name: Setup Argo CD
+    shell: bash
+    if: ${{ steps.tool-cache-argocd.outputs.cache-hit != 'true' }}
+    run: |-
+      curl -sSL -o /usr/local/bin/argocd https://github.com/argoproj/argo-cd/releases/download/${{ inputs.argocd_version }}/argocd-linux-amd64
+      chmod +x /usr/local/bin/argocd
+      argocd version
+    env:
+      ARGOCD_AUTH_TOKEN: ${{ inputs.argocd_token }}
+      ARGOCD_OPTS: "--grpc-web"
+      ARGOCD_SERVER: ${{ inputs.argocd_server }}
+  - name: Determine APP_NAME
+    shell: bash
+    run: |-
+      if [ "${{ inputs.app_name }}" == "" ]; then
+        APP_NAME="$(argocd app list -o json | jq -r '.[] | select(.spec.source.targetRevision=="${{ inputs.deployment_ref }}").metadata.name')"
+      else
+        APP_NAME="${{ inputs.app_name }}"
+      fi
+      echo "APP_NAME=${APP_NAME}" >> ${GITHUB_ENV}
+    env:
+      ARGOCD_AUTH_TOKEN: ${{ inputs.argocd_token }}
+      ARGOCD_OPTS: "--grpc-web"
+      ARGOCD_SERVER: ${{ inputs.argocd_server  }}
```

Comment:
Here I've already wondered in the previous per team action why we don't just supply the `app_name`. We know in 100% of the cases what the app is called, so why "overcomplicate" it by trying to find it and parse it.

Reply:
That is right 💯 , we can simplify this.